lib/site.js

```diff
@@ -58,34 +63,43 @@ module.exports = class Site {
 			await mkdirp(this.target);
 		}
 
-		let { tree, layout } = this;
+		let { tree, renderers, generateLayout } = this;
 		if(!tree || filepaths === null) {
 			tree = await this.load();
 		}
 
+		let layout = generateLayout(tree);
+		let pageRenderer = new PageRenderer({ renderers, layout });
+
 		let dirs = new Set();
-		let writes = [];
-		tree.forEach(async page => {
-			let filepath = page.targetPath(this.target);
+		let writes = tree.map(async page => {
```

Comment:
The old thing didn't do anything. The writes array was always empty, so you always waited on nothing.